android/app-newm/src/main/java/io/newm/wallet/connect/WalletConnectInitializer.kt

```diff
@@ -0,0 +1,61 @@
+package io.newm.wallet.connect
+
+import android.app.Application
+import com.walletconnect.android.Core
+import com.walletconnect.android.CoreClient
+import com.walletconnect.android.relay.ConnectionType
+import com.walletconnect.web3.modal.client.Modal
+import com.walletconnect.web3.modal.client.Web3Modal
+import com.walletconnect.web3.modal.presets.Web3ModalChainsPresets
+import io.newm.BuildConfig
+import io.newm.shared.public.constants.WalletConnectConstants
+import timber.log.Timber
+
+class WalletConnectInitializer(private val application: Application) {
+    private val TAG = "WalletConnectInitializer"
+    fun initialize() {
+        initializeCoreClient()
+        initializeWeb3Modal()
+    }
+
+    private fun initializeCoreClient() {
+        val connectionType = ConnectionType.AUTOMATIC
+        val relayUrl = "relay.walletconnect.com"
+        val serverUrl = "wss://$relayUrl?projectId=${BuildConfig.WALLET_CONNECT_PROJECT_ID}"
+        val appMetaData = Core.Model.AppMetaData(
+            name = "NEWM",
+            description = "Link your wallet to NEWM",
+            url = "https://newm.io/",
+            icons = listOf("https://developers.cardano.org/img/devblog/newm.png"),
```

Comment:
This file could get deleted right? Maybe use the one on ours. 
```suggestion
            icons = listOf("https://newm.io/wp-content/uploads/2022/06/NEWM_Logo-1536x1536.png"),
```